File: app/models/action.rb
class Action < ApplicationRecord

  has_many :job_actions

  def json_attributes
    {
      id: id,
      name: name,
      potency: potency,
      cost: cost,
      resource: resource,
      cast_time: cast_time,
      animation_lock: animation_lock,
      duration: duration,
      cooldown: cooldown,
      category: category,
      modifier: modifier,
      gcd_locked: gcd_locked,
      image_path: ActionController::Base.helpers.image_path("actions/#{image_name}")
    }
  end

end

File: app/assets/javascripts/stormblood/classes/action.js
function Action(obj) {
  this.id = obj.id;
  this.name = obj.name;
  this.potency = obj.potency;
  this.cost = obj.cost;
  this.resource = obj.resource;
  this.castTime = parseFloat(obj.cast_time);
  this.animationLock = parseFloat(obj.animation_lock);
  this.duration = parseFloat(obj.duration);
  this.cooldown = obj.cooldown;
  this.category = obj.category;
  this.modifier = parseFloat(obj.modifier);
  this.gcdLocked = obj.gcd_locked;
}
